File: src/main/java/net/benwoodworth/fastcraft/core/api/dependencies/event/FcEvent.java
package net.benwoodworth.fastcraft.core.api.dependencies.event;

import org.jetbrains.annotations.NotNull;

/**
 * Adapts an event from a native implementation.
 *
 * @param <TBase> the base event being adapted
 */
public interface FcEvent<TBase> {

    /**
     * The base of this event.
     *
     * @return the base of this event
     */
    @NotNull
    TBase getBaseEvent();

    /**
     * Raise the base event.
     */
    void raise();
}

File: src/main/kotlin/net/benwoodworth/fastcraft/core/api/dependencies/event/FcEvent.kt
package net.benwoodworth.fastcraft.core.api.dependencies.event

/**
 * Adapts an event from a native implementation.
 *
 * @param TBase the base event being adapted
 */
interface FcEvent<out TBase> {

    /** The base of this event. */
    val baseEvent: TBase

    /** Raise the base event. */
    fun raise()
}
